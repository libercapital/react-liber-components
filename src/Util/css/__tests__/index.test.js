import { generateMinMax, disableOnTest } from '..';

describe('Test css functions', () => {
  describe('generateMinMax()', () => {
    it('should return css values for width', () => {
      expect(generateMinMax('width', '40px')).toMatchSnapshot();
    });
    it('should return css values for height', () => {
      expect(generateMinMax('height', '40px').join('')).toMatchSnapshot();
    });
  });

  describe('disableOnTest()', () => {
    it('should not return css when testing', () => {
      expect(disableOnTest`css test`).toEqual(null);
    });

    it('should return css on any other case', () => {
      const { env } = process;

      jest.resetModules();
      process.env = { ...env };
      delete process.env.NODE_ENV;

      process.env.NODE_ENV = 'development';
      expect(disableOnTest`css test`).toContain('css test');
      process.env = env;
    });
  });
});
