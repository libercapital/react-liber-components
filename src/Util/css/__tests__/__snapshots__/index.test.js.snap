// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test css functions generateMinMax() should return css values for height 1`] = `"min-height:40px;max-height:40px;"`;

exports[`Test css functions generateMinMax() should return css values for width 1`] = `
Array [
  "min-",
  "width",
  ":",
  "40px",
  ";max-",
  "width",
  ":",
  "40px",
  ";",
]
`;
